corehq/apps/domain/models.py

```diff
@@ -585,6 +592,15 @@ def get_by_organization_and_hrname(cls, organization, hr_name):
         return result
 
     @classmethod
+    def get_by_alias(cls, domain_alias):
```

Comment:
This should use the same pattern as `get_by_name`.  If you make the couch view's have the same key structure you should be able to use exactly the same function just with a different view name.